### DISCUSSION TITLE
Bypass Internet Censorship feature not working

### DISCUSSION THREAD

**Discussion by greykaizen** [celzero/rethink-app#463] [General]
I tried the **DNS and Firewall (default)** , and the sites which are banned in my country are not opening.
I tried cloudflare's **1.1.1.1** app with warp connection it works fine and bypass internet censorship too. 
Is there a solution to that problem in rethinkdns because i tried different dns mention in options there, also use dns without firewall, still no solution.

**Comment by ignoramous** [comment]
RethinkDNS is _not_ a VPN, so it is limited in amount of censorship it can bypass. Specifically, RethinkDNS bypasses stateless firewalls doing SNI (server name identification) based blocks (Turkmenistan, India, etc). It cannot bypass firewalls doing IP blocks (Russia) or stateful firewall doing sophisticated DPI (deep packet inspection) (Iran, China, etc).

**Reply by greykaizen** [reply]
I understand that ... 

Now the issue is dns sometimes work and sometimes it shows connected but dont work ...
I was using an app in the morning using rethinkdns, i wasn't using my phone so disconnected the dns, later on when i connected my device to rethinkdns it wasn't working but showing status connected/protected.

I reinstalled the app, it started working but then disconnected it for a while later reconnected and it stopped working, facing the issue right now too

**Reply by ignoramous** [reply]
Strange.

> I reinstalled the app, it started working but then disconnected it for a while later reconnected and it stopped working, facing the issue right now too

Can you please answer these:

1. What ROM / OS are you on?
2. Is your device *losing* internet connectivity? Or, is not able to unblock censored websites?
3. What does RethinkDNS homescreen status say, "waiting" (in red colour), "waiting" (blue), "failing" (red), "starting" (blue), "protected" (blue)?
4. What do you see in the DNS and Network logs when RethinkDNS "stops working".

Thanks.

**Reply by greykaizen** [reply]
1. Android 11, realme 6 , 0.5.3h latest Version.

2. No, internet is fine ... I just disconnected rethinkdns and connected later on but it was not working.

3. Its shows **protected**

4. In log section, its showing multiple enteries. When the issue occurs the logs section stop adding more enteries

While writing this reply, i tried **DNS over HTTPS** and **DNS proxy** with multiple options given there but it wasn't even reloading site for github, then i use cloudflare's **1.1.1.1** app to see if that happens there but everything seems to work just fine.

**Reply by ignoramous** [reply]
Could it be that moving between Wifi and Mobile network is the difference? Or, rather RethinkDNS doesn't work on *some* Wifis / Mobile networks?

I am confused, but would help if you can clear it up: Is the issue that RethinkDNS doesn't bypass blocked sites, or that RethinkDNS doesn't work at all no matter which DNS server you use (on all networks: Mobile and Wifi)?

**Reply by greykaizen** [reply]
I'm using Wifi, no mobile data
Secondly it _wasn't working at all_ as i said when i connected my device to rethink it wasn't even loading github site, then i switched to 1.1.1.1 app the internet start working. So internet is not the problem.

I clear cache and data and then went to settings **VPN** disconnected RethinkDns and then connected it through app and it start working again. I'll probably disconnect it in the night and i dont know if its gonna work in the morning or not

**Reply by ignoramous** [reply]
Ah, gotcha. That's one annoying issue, alright.

So you're seeing RethinkDNS drop internet connectivity altogether. Btw, doesn't simply stop/start RethinkDNS app work? I'd wager it should. Can you confirm?

We have seen reports of such issues in v053c/d/e versions of the app (see #248), but I am surprised a similar issue is still prevalent on Realme/Android11.

**Reply by greykaizen** [reply]
Sometimes ... I mean rethink has a bug, once connected, when clicked **stop** it will only show **waiting**, it doesn't disconnect fully, in notification panel it was showing rethink disconnected but the app was showing **waiting** status other dns so also don't work till i disconnect rethink from Android settings.

**Reply by ignoramous** [reply]
Btw: We've improved the statuses (`waiting`/`protected`/`starting`/etc) we show on the app. #552

**Comment by ignoramous** [comment]
We are working on other bypasses for "stateless firewalls": celzero/firestack#1

And adding support for WireGuard VPN: #52 

If possible, we also plan to add support for Socksifiers: #51
